### ISSUE TITLE
encore amélio

### ISSUE THREAD

**Issue by Rapidhands** [dakhama-mehdi/Modern_ActiveDirectory#6]
re Medhi
J'ai regardé le code  (ex ici function_HTMLMultiPage.ps1). Dans ton code tu as 100 les mêmes cmdlets (Ok), mais avec 100 fois les mêmes  paramètres (ex. New-HTMLText -Text $userinactive -Alignment justify -FontSize 25 -FontWeight bold )
tu aurais tout intérêt à te simplifier le code en utilisant en début de script un plat pour définir les valeurs par défaut de ces cmdlets dans la variable PSDefaultParameterValues. Ainsi, tu n'aurais qu'à saisir a cmdlet et sans paramètre spécifié cela appliquerait ceux du PDDefaultParameterValues. De plus,si tu veux en changer 1, tu n'as à le changer qu'une seule fois dans la variable et tu n'as pas à te coltiner tout le code. 

un exemple de splat pour cela:
```
$PSDefaultParameterValues = @{
                                    '*:Encoding' = 'UTF8'
                                    'Install-Module:Scope' = 'CurrentUser'
                                    'Test-Connection:Count' = '1' 
                                    'Export-Csv:NoTypeInformation' = $true
                                    'Export-Csv:delimiter' = ";"
                                    'ConvertTo-Csv:NoTypeInformation' = $true
                                    'Get-WinEvent:LogName'='System'
                                    }
```

cordialement
Olivier

**Comment by dakhama-mehdi** [comment]
Oui, le code HTML est a revoir et peu bcp etre amélioré et simplifié, au départ je l'ai construis au millimètre fallait calcule la largeur, hauteur ne pas décalé et hop on revoit tout, j'étais aussi beaucoup aidé par le développeur du module qui a revu lui aussi pas mal de chose.
Je vais essayer de réduire cela dans des sous fonctions mais je sais pas si ca marchera ou pas, car il a une méthode native

**Comment by dakhama-mehdi** [comment]
Le code HTML est un appel d'une fonction d'un autre module, peut etre que je tenterai une function qui me retourne la ligne entiere et je lui envoie que les parametres, mais je ne pense pas que le module html à été pensé comme ça, j'ai déja tenté des modifications mais toujours déformer en résultat.

**Comment by Rapidhands** [comment]
ce n'est pas ça dont je te parle. Tu passes - et tu n'as pas le choix- xx fois les mêmes cmdlets, mais dans 95% du temps avec les mêmes paramètres et les mêmes valeurs. Mets ces paramètres et ces valeurs dans le $PSDefaultParameterValue et tu n'auras pas besoin de les écrire, et ça rend le code plus lisible. Si tu ne sais pas ce qu'est cette variable dont je parle, c'est une variable prédéfinie, une recherche sur Internet t'en dira plus
